bigquery/google/cloud/bigquery/table.py

```diff
@@ -1334,8 +1339,26 @@ def _to_dataframe_tabledata_list(self, dtypes):
         """Use (slower, but free) tabledata.list to construct a DataFrame."""
         column_names = [field.name for field in self.schema]
         frames = []
+
+        # report progress if tqdm installed
+        pbar = None
+        if tqdm is not None:
+            try:
+                pbar = tqdm.tqdm(
+                    desc="Downloading", total=self.total_rows, unit="rows"
+                )
+            except (KeyError, TypeError):
+                # tqdm error
+                pass
+
         for page in iter(self.pages):
             frames.append(self._to_dataframe_dtypes(page, column_names, dtypes))
+
+            if pbar is not None:
+                pbar.total = pbar.total or self.total_rows
+                # update progress bar with number of rows in last frame
```

Comment:
This comment is unnecessary, since it just states what is happening in the next line. Comments in this codebase should state *why* or explain something that looks "wrong".

This comment indicates to me that we need to rename `pbar` to `progress_bar`. Also, we could add some intermediate variables above to make this line more self-explanatory.

```
frames.append(self._to_dataframe_dtypes(page, column_names, dtypes))

     |
turns into
     |
     V

current_frame = self._to_dataframe_dtypes(page, column_names, dtypes)
frames.append(current_frame)
...
    progress_bar.update(len(current_frame))
```

Reply:
Awesome feedback, thank you :)